ui/apps/dashboard/src/app/api/chat/route.ts

```diff
@@ -0,0 +1,85 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { auth } from '@clerk/nextjs/server';
+import { z } from 'zod';
+
+import { inngest } from '../inngest/client';
+
+// Zod schema for UserMessage
+const userMessageSchema = z.object({
+  id: z.string().uuid('Valid message ID is required'),
+  content: z.string().min(1, 'Message content is required'),
+  role: z.literal('user'),
+  state: z.record(z.unknown()).optional(),
+  clientTimestamp: z.coerce.date().optional(), // ✅ Coerce string to Date object
+  systemPrompt: z.string().optional(),
+});
+
+// Zod schema for request body validation
+const chatRequestSchema = z.object({
+  userMessage: userMessageSchema,
+  threadId: z.string().uuid().optional(),
+  userId: z.string().optional(),
+  channelKey: z.string().optional(), // NEW: Support channelKey for flexible subscriptions
+  history: z.array(z.any()).optional(), // TODO: define a more specific schema for history items
+});
+
+export async function POST(req: NextRequest) {
+  try {
+    // Authenticate the user using Clerk
```

Comment:
Is there middleware we can use to automatically auth all endpoints? Putting it on each endpoint scares me a little, since an engineer might forget it or do it wrong

Reply:
We don't want that for the Inngest functions and can figure that out in a followup.